include/ringbuffer.h

```diff
@@ -0,0 +1,132 @@
+#include <inttypes.h>
+
+/**
+ * @file
+ * Prototypes and structures for the ring buffer module.
+ */
+
+#ifndef RINGBUFFER_H
+#define RINGBUFFER_H
+
+/**
+ * The size of a ring buffer.
+ * Due to the design only <tt> RING_BUFFER_SIZE-1 </tt> items
+ * can be contained in the buffer.
+ * The buffer size must be a power of two.
+*/
+#define RING_BUFFER_SIZE 128
+
+#if (RING_BUFFER_SIZE & (RING_BUFFER_SIZE - 1)) != 0
+#error "RING_BUFFER_SIZE must be a power of two"
+#endif
+
+/**
+ * The type which is used to hold the size
+ * and the indicies of the buffer.
+ * Must be able to fit \c RING_BUFFER_SIZE .
+ */
+typedef uint8_t ring_buffer_size_t;
```

Comment:
Don't do this: https://www.kernel.org/doc/html/v4.10/process/coding-style.html#typedefs